subsys/net/lib/sockets/sockets_can.c

```diff
@@ -98,19 +98,20 @@ static void zcan_received_cb(struct net_context *ctx, struct net_pkt *pkt,
 	int i;
 
 	for (i = 0; i < ARRAY_SIZE(receivers); i++) {
-		struct zcan_frame *zframe =
-			(struct zcan_frame *)net_pkt_data(pkt);
-		struct can_frame frame;
+		struct can_frame frame = *(struct can_frame *)net_pkt_data(pkt);
```

Comment:
As you can see, a `can_frame` is pushed into the message queue in socket_can_generic.h. 
We are receiving can_frame, because they can be filtered for error frames and zcan_frame can't.